### PR TITLE
Add support for `level` in `PatchWSIDataset`

### LINKED ISSUE #4034: Support for chosen level in `PatchWSIDataset`
**Is your feature request related to a problem? Please describe.**
Following up on #3995, user likes to be able to define the whole slide image `level` when using `PatchWSIDataset` in order to load patches at that `level`.

**Describe the solution you'd like**
For the solution, we should consider that the `location` is always refer to the coordinates at `level=0`.

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #4034 

### Description
Add supports in `PatchWSIDataset` to accept `level` as an argument and extract patches from the whole slide images at that `level`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @drbeh ,

Thanks for your quick update according to user's feedback, I have 2 comments:
1. Should we also add the `level` arg to other datasets in the file?
Like `SmartCachePatchWSIDataset`, `MaskedInferenceWSIDataset`.
3. As `WSIReader` also supports `**kwargs`, could you please help add it to the datasets?
Then I think it can help support more similar requirements in future.

What do you think?

Thanks.

**Comment by drbeh** [comment]
Thank you @Nic-Ma,  it was very helpful comment.

File: monai/apps/pathology/data/datasets.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ class PatchWSIDataset(Dataset):
         region_size: the size of regions to be extracted from the whole slide image.
         grid_shape: the grid shape on which the patches should be extracted.
         patch_size: the size of patches extracted from the region on the grid.
+        level: the level at which the patches to be extracted (default to 0).
```
Comment (Nic-Ma) [inline monai/apps/pathology/data/datasets.py:35]: As you changed the arg to `kwargs`, could you please help update the doc-string to `kwargs` here?
And also add doc-string to other datasets?

Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**